kolena/_experimental/dataset/_dataset.py

```diff
@@ -113,15 +121,52 @@ def _to_deserialized_dataframe(df: pd.DataFrame, column: str) -> pd.DataFrame:
     return result
 
 
-def register_dataset(name: str, df: pd.DataFrame) -> None:
+def _upload_dataset_chunk(df: pd.DataFrame, load_uuid: str, id_fields: List[str]) -> None:
+    df_serialized_datapoint = _to_serialized_dataframe(df, column=COL_DATAPOINT)
+    df_serialized_datapoint_id_object = _to_serialized_dataframe(df[sorted(id_fields)], column=COL_DATAPOINT_ID_OBJECT)
+    df_serialized = pd.concat([df_serialized_datapoint, df_serialized_datapoint_id_object], axis=1)
+
+    upload_data_frame(df=df_serialized, batch_size=BatchSize.UPLOAD_RECORDS.value, load_uuid=load_uuid)
+
+
+def load_dataset(name: str) -> Optional[EntityData]:
+    response = krequests.put(Path.LOAD_DATASET, json=asdict(LoadDatasetByNameRequest(name=name)))
+    if response.status_code == requests.codes.not_found:
+        return None
+
+    response.raise_for_status()
+
+    return from_dict(EntityData, response.json())
+
+
+def register_dataset(name: str, df: Union[Iterator[pd.DataFrame], pd.DataFrame], id_fields: List[str]) -> None:
```

Comment:
Non blocking: Only highlighting one limitation is that id uniqueness check would not be complete if users pass an iterator.

Reply:
We do have dedup on the temp table on server side, the behavior would be different. Would this be acceptable?

Reply:
I'm good with it.